SimplifiedTests/NYPLAxisTests/NYPLAxisDownloadProgressTests.swift

```diff
@@ -0,0 +1,312 @@
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: chapter2URL), 0.95/2)
+
+    sut.addFixedWeightTask(with: containerURL, weight: 0.05)
+    XCTAssertEqual(progressListener.currentProgress, 0.05)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: licenseURL), 0.05)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: containerURL), 0.05)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: chapter1URL), 0.45)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: chapter2URL), 0.45)
+
+    sut.addFlexibleWeightTask(with: chapter3URL)
+    XCTAssertEqual(progressListener.currentProgress, 0.05)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: licenseURL), 0.05)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: containerURL), 0.05)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: chapter1URL), 0.3)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: chapter2URL), 0.3)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: chapter3URL), 0.3)
+
+    sut.addFixedWeightTask(with: encryptionURL, weight: 0.05)
+    XCTAssertEqual(progressListener.currentProgress, 0.05)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: licenseURL), 0.05)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: containerURL), 0.05)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: encryptionURL), 0.05)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: chapter1URL), 0.85/3)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: chapter2URL), 0.85/3)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: chapter3URL), 0.85/3)
+
+    // 10 % completed
+    sut.didFinishTask(with: encryptionURL)
+    XCTAssertEqual(progressListener.currentProgress, 0.1)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: licenseURL), 0.05)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: containerURL), 0.05)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: encryptionURL), 0.05)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: chapter1URL), 0.85/3)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: chapter2URL), 0.85/3)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: chapter3URL), 0.85/3)
+
+    // 15 % completed
+    sut.didFinishTask(with: containerURL)
+    XCTAssertEqual(progressListener.currentProgress, 0.15)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: licenseURL), 0.05)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: containerURL), 0.05)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: encryptionURL), 0.05)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: chapter1URL), 0.85/3)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: chapter2URL), 0.85/3)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: chapter3URL), 0.85/3)
+
+    // ~43% completed
+    sut.didFinishTask(with: chapter2URL)
+    XCTAssertEqual(progressListener.currentProgress, (0.15 + (0.85/3)).roundedToTwoDecimalPlaces)
```

Comment:
great test -- but if you round it why not just using `0.43`?

Reply:
You're right. It is 43%.